cmd/juju/application/deployer/charm.go

```diff
@@ -252,7 +253,20 @@ func (d *deployCharm) deploy(
 		Resources:        ids,
 		EndpointBindings: d.bindings,
 	}
-	return errors.Trace(deployAPI.Deploy(args))
+
+	err = deployAPI.Deploy(args)
+	if err == nil {
+		return nil
+	}
+
+	if pErr, ok := errors.Cause(err).(*params.Error); ok {
```

Comment:
How about moving this check into the client and have that return an `errors.AlreadyExists` that can be tested for and wrapped here? I would prefer we didn't deal with `params.Error` at this level.

Reply:
Sure makes sense